zebra/zebra_nhg.h

```diff
@@ -190,6 +190,9 @@ enum nhg_type {
 
 #define PROTO_OWNED(NHE) (NHE->id >= ZEBRA_NHG_PROTO_LOWER)
 
+/* return TRUE if the NHG is Singleton */
+#define ZEBRA_NHG_IS_SINGLETON(NHE) (NHE->nhg.nexthop && !NHE->nhg.nexthop->next)
```

Comment:
maybe the core of this should be in the lib/nexthop_group module, since that's really what's being tested?